### PR TITLE
Use Microsoft.Artifacts.Authentication in credential provider

### PR DESCRIPTION
**Pull request by JohnSchmeichel**
Adding a reference to the Microsoft.Artifacts.Authentication project and removing all associated code from the credential provider that it replaces. Also includes a few bug fixes present in the credential provider that were not yet ported to the library.

### CODE REVIEW

File: src/Authentication/Resources.resx [hunk 1]
```diff
@@ -150,4 +150,7 @@
   <data name="MsalUsingWamBroker" xml:space="preserve">
     <value>MSAL using WAM Broker</value>
   </data>
+  <data name="MsalNotUserInteractive" xml:space="preserve">
+    <value>MSAL not user interactive</value>
```
Comment (johnterickson) [inline src/Authentication/Resources.resx:154]: "Cannot use interactive auth in a non-interactive app"

Comment (johnterickson) [reply src/Authentication/Resources.resx:154]: If I am interpreting that right ;)

File: src/Authentication/AzureArtifacts.cs [hunk 2]
```diff
@@ -12,10 +12,14 @@ namespace Microsoft.Artifacts.Authentication;
 public static class AzureArtifacts
 {
     public const string ClientId = "d5a56ea4-7369-46b8-a538-c370805301bf";
+    private const string LegacyClientId = "872cd9fa-d31f-45e0-9eab-6e460a02d1f1";
```
Comment (johnterickson) [inline src/Authentication/AzureArtifacts.cs:15]: Add comments with the "name' of each

File: src/Authentication/AzureArtifacts.cs [hunk 3]
```diff
@@ -12,10 +12,14 @@ namespace Microsoft.Artifacts.Authentication;
...
 
     public static PublicClientApplicationBuilder CreateDefaultBuilder(Uri authority)
     {
-        var builder = PublicClientApplicationBuilder.Create(AzureArtifacts.ClientId)
+        // Azure Artifacts is not yet present in PPE, so revert to the old app in that case
+        bool ppe = authority.Host.Equals("login.windows-ppe.net", StringComparison.OrdinalIgnoreCase);
+
+        var builder = PublicClientApplicationBuilder.Create(!ppe ? AzureArtifacts.ClientId : AzureArtifacts.LegacyClientId)
```
Comment (johnterickson) [inline src/Authentication/AzureArtifacts.cs:22]: Can we not not in the ternary? Maybe bool prod =!...

File: CredentialProvider.Microsoft.VSIX/Microsoft.CredentialProvider.swr [hunk 4]
```diff
@@ -7,6 +7,15 @@ folder InstallDir:\Common7\IDE\CommonExtensions\Microsoft\NuGet\Plugins\Credenti
   file source=$(PluginBinPath)\CredentialProvider.Microsoft.exe
   file source=$(PluginBinPath)\CredentialProvider.Microsoft.exe.config
   file source=$(PluginBinPath)\CredentialProvider.Microsoft.pdb
+  file source=$(PluginBinPath)\Microsoft.Artifacts.Authentication.dll
+  file source=$(PluginBinPath)\Microsoft.Artifacts.Authentication.pdb
```
Comment (johnterickson) [inline CredentialProvider.Microsoft.VSIX/Microsoft.CredentialProvider.swr:11]: do we want to embed the pdb?

**Review by johnterickson — CHANGES_REQUESTED** [review]
:clock1:

File: CredentialProvider.Microsoft.Tests/CredentialProviders/Vsts/AccountPriorityTests.cs [hunk 5]
```diff
@@ -1,114 +0,0 @@
...
-            new List<IAccount> { ContosoUser, MsaUser, FabrikamUser },
-            new List<IAccount> { ContosoUser, FabrikamUser, MsaUser },
-            new List<IAccount> { MsaUser, ContosoUser, FabrikamUser },
-            new List<IAccount> { MsaUser, FabrikamUser, ContosoUser },
-            new List<IAccount> { FabrikamUser, MsaUser, ContosoUser },
-            new List<IAccount> { FabrikamUser, ContosoUser, MsaUser },
-        };
-
-        [TestMethod]
-        public void MsaMatchesMsa()
```
[the remark this answers is not in the document]
Comment (JohnSchmeichel) [reply CredentialProvider.Microsoft.Tests/CredentialProviders/Vsts/AccountPriorityTests.cs:59]: These were already moved under src/Authentication.Tests: https://github.com/microsoft/artifacts-credprovider/blob/3645b391081f97c3d09a542c22e851a401769a68/src/Authentication.Tests/AccountPriorityTests.cs

**Comment by JohnSchmeichel** [comment]
> 🕐

Addressed all comments

**Review by johnterickson — APPROVED** [review]
:shipit:

**Pull request merged**